examples/inference/HF_inference.py

```diff
@@ -137,23 +155,20 @@ def run_all(pp_ranks, args):
     parser.add_argument('--visualize', type=int, default=1, choices=[0, 1])
     parser.add_argument('--pp_group_size', type=int, default=int(os.getenv("WORLD_SIZE", 4)))
     parser.add_argument('--auto_split', type=str, default="equal_size")
+    parser.add_argument('--index_filename', type=str, default=None, help="The director of model's index.json file")
 
     args = parser.parse_args()
 
     assert args.world_size % args.pp_group_size == 0
 
     # Main process loads model
     print(f"Loading model {args.model_name}")
-    if 't5' in args.model_name:
-        model = AutoModelForSeq2SeqLM.from_pretrained(args.model_name, use_cache=False)
-    if 'opt' in args.model_name:
-        model = OPTModel.from_pretrained(args.model_name, use_cache=False)
-    if 'bloom' in args.model_name:
-        model = BloomModel.from_pretrained(args.model_name, use_cache=False)
-    if 'regnet' in args.model_name:
-        feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/regnet-y-10b-seer")
-        model = RegNetModel.from_pretrained("facebook/regnet-y-10b-seer")
-        args.feature_extractor = feature_extractor
+    if args.index_filename is not None:
+        with init_empty_weights():
```

Comment:
Thanks for your comments. It would be great if you can add the init_empty_weights to PT. In this case, I only import accelerate in the Huggingface example, it would not be a problem since accelerate is belong to Huggingface and many examples in transformers also used accelerate.